src/Durable/ActivityInvocationTracker.cs

```diff
@@ -75,6 +86,16 @@ internal class ActivityInvocationTracker
             var allTasksCompleted = completedEvents.Count == tasksToWaitFor.Count;
             if (allTasksCompleted)
             {
+                var newOrchestrationStart = context.History.FirstOrDefault(
+                    (e) => e.EventType == HistoryEventType.OrchestratorStarted &&
+                    e.Timestamp > context.CurrentUtcDateTime
```

Comment:
Please format this in a way that makes it clear that both `e.EventType == ...` and `e.Timestamp > ...` belong to the same lambda body. I would just insert a few spaces so that `e.Timestamp` is directly below `e.EventType`.